api/stream/stream.go

```diff
@@ -71,29 +75,42 @@ func (a *API) Notify(userID uint, msg *model.MessageExternal) {
 	a.lock.RLock()
 	defer a.lock.RUnlock()
 	if clients, ok := a.clients[userID]; ok {
-		for _, c := range clients {
-			c.write <- msg
+		if msg.Clients!=nil && len(msg.Clients)>0{
+			for _,token := range msg.Clients{
+				if c,ok := clients[token];ok {
+					c.write <- msg
+				}
+			}
+		}else{
+			for _, c := range clients {
+				c.write <- msg
+			}	
 		}
+		
 	}
 }
 
 func (a *API) remove(remove *client) {
 	a.lock.Lock()
 	defer a.lock.Unlock()
 	if userIDClients, ok := a.clients[remove.userID]; ok {
-		for i, client := range userIDClients {
+		for key, client := range userIDClients {
 			if client == remove {
-				a.clients[remove.userID] = append(userIDClients[:i], userIDClients[i+1:]...)
+				delete(a.clients[remove.userID],key)
+				//a.clients[remove.userID] = append(userIDClients[:i], userIDClients[i+1:]...)
 				break
 			}
 		}
 	}
 }
 
-func (a *API) register(client *client) {
+func (a *API) register(cliente *client) {
```

Comment:
Why rename this variable?

Reply:
because when i use the variable as client, the line 111 sent me an error, maybe the compiler got confused with the structure and the variable